src/ShopBundle/objects/TShopBasket/TShopBasketCore.class.php

```diff
@@ -640,6 +640,9 @@ public function updateCustomData(string $basketIdentifier, array $customData): v
             return;
         }
         $item->setCustomData($customData);
+
+        // this may result in items with the same basket ident - in that case we should merge them
+        $this->oBasketArticles->mergeIdenticalBasketItems();
```

Comment:
The method updateCustomData seems quite innocent.

Why should it result in items being added to the basket?

Reply:
not at all. The basketIdentKey of an item always depends on its custom data. When a users changes the custom data of an item, they also change the basketIdentKey - which then may result in having two items with the same key.

In fact, this is exactly the problem - a user has product A with some custom data foo=bar  and the same product A with some custom data foo=baz. When the user then changes foo=baz to foo=bar in the second product, the result will be product A being twice in the basket with identical basketIdentKeys. These must now be merged

Reply:
Hm, I think what I find unfortunate here:

setting custom data on an entry breaks something and requires the list to be fixed (and thus the method mergeIdenticalBasketItems there is public but needs to be called here).

Reply:
ok - that makes sense. I moved alle the logic into the article list - so we no longer need the public method mergeIdenticalBasketItems